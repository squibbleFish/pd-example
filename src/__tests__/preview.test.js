import React from 'react';
import { shallow } from 'enzyme';

import Preview from '../components/preview';

describe('Render a preview screen of themes', () => {
  it('should render a preview', function () {
    const props = {
      location: {
        state: {
          theme: {
            created: "2018-01-19T21:58:28.528Z",
            description: "A theme ready for action, with a hero cover image and a sidebar.",
            guid: "e11ca530-fd63-11e7-a5be-e53eb9f02238",
            handle: "sidekick",
            last_updated: "2018-01-19T21:58:31.792Z",
            main: "73ac0428-8400-4202-abde-7ca1bf46ad09",
            owner: "_default",
            title: "Sidekick",
            type: "theme",
          }
        }
      }
    };
    const wrapper = shallow(
      <Preview { ...props } />
    );
    expect(wrapper).toMatchSnapshot();
  });
});