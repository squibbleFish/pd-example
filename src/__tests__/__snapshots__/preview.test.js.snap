// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render a preview screen of themes should render a preview 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Preview
    location={
        Object {
            "state": Object {
              "theme": Object {
                "created": "2018-01-19T21:58:28.528Z",
                "description": "A theme ready for action, with a hero cover image and a sidebar.",
                "guid": "e11ca530-fd63-11e7-a5be-e53eb9f02238",
                "handle": "sidekick",
                "last_updated": "2018-01-19T21:58:31.792Z",
                "main": "73ac0428-8400-4202-abde-7ca1bf46ad09",
                "owner": "_default",
                "title": "Sidekick",
                "type": "theme",
              },
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <section
          className="flex"
>
          <h4>
                    Theme Details
          </h4>
          <p>
                    <span
                              className="bold"
                    >
                              Handle: 
                    </span>
                    sidekick
          </p>
          <p>
                    <span
                              className="bold"
                    >
                              Created: 
                    </span>
                    January 19th 2018, 1:58:28
                    <br />
                    <span
                              className="bold"
                    >
                              Last Updated: 
                    </span>
                    January 19th 2018, 1:58:31
          </p>
          <p
                    className="bold"
          >
                    A theme ready for action, with a hero cover image and a sidebar.
          </p>
          <p>
                    <span
                              className="bold"
                    >
                              GUID: 
                    </span>
                    e11ca530-fd63-11e7-a5be-e53eb9f02238
          </p>
</section>,
        <section
          className="flex frame"
>
          <iframe
                    frameBorder="0"
                    src="https://pagedip.com/theme/_default/sidekick"
          />
</section>,
      ],
      "className": "theme-preview",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h4>
              Theme Details
</h4>,
            <p>
              <span
                            className="bold"
              >
                            Handle: 
              </span>
              sidekick
</p>,
            <p>
              <span
                            className="bold"
              >
                            Created: 
              </span>
              January 19th 2018, 1:58:28
              <br />
              <span
                            className="bold"
              >
                            Last Updated: 
              </span>
              January 19th 2018, 1:58:31
</p>,
            <p
              className="bold"
>
              A theme ready for action, with a hero cover image and a sidebar.
</p>,
            <p>
              <span
                            className="bold"
              >
                            GUID: 
              </span>
              e11ca530-fd63-11e7-a5be-e53eb9f02238
</p>,
          ],
          "className": "flex",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Theme Details",
            },
            "ref": null,
            "rendered": "Theme Details",
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="bold"
>
                  Handle: 
</span>,
                "sidekick",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Handle: ",
                  "className": "bold",
                },
                "ref": null,
                "rendered": "Handle: ",
                "type": "span",
              },
              "sidekick",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="bold"
>
                  Created: 
</span>,
                "January 19th 2018, 1:58:28",
                <br />,
                <span
                  className="bold"
>
                  Last Updated: 
</span>,
                "January 19th 2018, 1:58:31",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Created: ",
                  "className": "bold",
                },
                "ref": null,
                "rendered": "Created: ",
                "type": "span",
              },
              "January 19th 2018, 1:58:28",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Last Updated: ",
                  "className": "bold",
                },
                "ref": null,
                "rendered": "Last Updated: ",
                "type": "span",
              },
              "January 19th 2018, 1:58:31",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "A theme ready for action, with a hero cover image and a sidebar.",
              "className": "bold",
            },
            "ref": null,
            "rendered": "A theme ready for action, with a hero cover image and a sidebar.",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="bold"
>
                  GUID: 
</span>,
                "e11ca530-fd63-11e7-a5be-e53eb9f02238",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "GUID: ",
                  "className": "bold",
                },
                "ref": null,
                "rendered": "GUID: ",
                "type": "span",
              },
              "e11ca530-fd63-11e7-a5be-e53eb9f02238",
            ],
            "type": "p",
          },
        ],
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <iframe
            frameBorder="0"
            src="https://pagedip.com/theme/_default/sidekick"
/>,
          "className": "flex frame",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "frameBorder": "0",
            "src": "https://pagedip.com/theme/_default/sidekick",
          },
          "ref": null,
          "rendered": null,
          "type": "iframe",
        },
        "type": "section",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <section
            className="flex"
>
            <h4>
                        Theme Details
            </h4>
            <p>
                        <span
                                    className="bold"
                        >
                                    Handle: 
                        </span>
                        sidekick
            </p>
            <p>
                        <span
                                    className="bold"
                        >
                                    Created: 
                        </span>
                        January 19th 2018, 1:58:28
                        <br />
                        <span
                                    className="bold"
                        >
                                    Last Updated: 
                        </span>
                        January 19th 2018, 1:58:31
            </p>
            <p
                        className="bold"
            >
                        A theme ready for action, with a hero cover image and a sidebar.
            </p>
            <p>
                        <span
                                    className="bold"
                        >
                                    GUID: 
                        </span>
                        e11ca530-fd63-11e7-a5be-e53eb9f02238
            </p>
</section>,
          <section
            className="flex frame"
>
            <iframe
                        frameBorder="0"
                        src="https://pagedip.com/theme/_default/sidekick"
            />
</section>,
        ],
        "className": "theme-preview",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4>
                Theme Details
</h4>,
              <p>
                <span
                                className="bold"
                >
                                Handle: 
                </span>
                sidekick
</p>,
              <p>
                <span
                                className="bold"
                >
                                Created: 
                </span>
                January 19th 2018, 1:58:28
                <br />
                <span
                                className="bold"
                >
                                Last Updated: 
                </span>
                January 19th 2018, 1:58:31
</p>,
              <p
                className="bold"
>
                A theme ready for action, with a hero cover image and a sidebar.
</p>,
              <p>
                <span
                                className="bold"
                >
                                GUID: 
                </span>
                e11ca530-fd63-11e7-a5be-e53eb9f02238
</p>,
            ],
            "className": "flex",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Theme Details",
              },
              "ref": null,
              "rendered": "Theme Details",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="bold"
>
                    Handle: 
</span>,
                  "sidekick",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Handle: ",
                    "className": "bold",
                  },
                  "ref": null,
                  "rendered": "Handle: ",
                  "type": "span",
                },
                "sidekick",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="bold"
>
                    Created: 
</span>,
                  "January 19th 2018, 1:58:28",
                  <br />,
                  <span
                    className="bold"
>
                    Last Updated: 
</span>,
                  "January 19th 2018, 1:58:31",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Created: ",
                    "className": "bold",
                  },
                  "ref": null,
                  "rendered": "Created: ",
                  "type": "span",
                },
                "January 19th 2018, 1:58:28",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Last Updated: ",
                    "className": "bold",
                  },
                  "ref": null,
                  "rendered": "Last Updated: ",
                  "type": "span",
                },
                "January 19th 2018, 1:58:31",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "A theme ready for action, with a hero cover image and a sidebar.",
                "className": "bold",
              },
              "ref": null,
              "rendered": "A theme ready for action, with a hero cover image and a sidebar.",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="bold"
>
                    GUID: 
</span>,
                  "e11ca530-fd63-11e7-a5be-e53eb9f02238",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "GUID: ",
                    "className": "bold",
                  },
                  "ref": null,
                  "rendered": "GUID: ",
                  "type": "span",
                },
                "e11ca530-fd63-11e7-a5be-e53eb9f02238",
              ],
              "type": "p",
            },
          ],
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <iframe
              frameBorder="0"
              src="https://pagedip.com/theme/_default/sidekick"
/>,
            "className": "flex frame",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "frameBorder": "0",
              "src": "https://pagedip.com/theme/_default/sidekick",
            },
            "ref": null,
            "rendered": null,
            "type": "iframe",
          },
          "type": "section",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
